addons/account_check_printing/models/account_payment.py

```diff
@@ -61,6 +61,7 @@ def _constrains_check_number(self):
                AND payment.id IN %(ids)s
                AND move.state = 'posted'
                AND other_move.state = 'posted'
+               AND payment.check_number IS NOT NULL
```

Comment:
don't need.

but `'ids': tuple(self.ids),` --> `'ids': tuple(payment_check.ids),`

Reply:
@fmdl 

Thank you for review

I just updated the self.ids -> payment_checks.ids


> don't need.

Do you refer to `IS NOT NULL`?

Without this IS NOT NULL the output of the analyze of the query is:

    Planning Time: 3.354 ms
    Execution Time: 2514.660 ms

With `IS NOT NULL`

```diff
    AND other_move.state = 'posted';
+    AND payment.check_number IS NOT NULL
```

The output is:

    Planning Time: 3.216 ms
    Execution Time: 0.140 ms

Could you help me to understand why it is not needed, please?

Reply:
The query planner shows the following differences:

```patch
-    AND payment.check_number IS NOT NULL
```
`->  Nested Loop  (cost=1.43..2577.12 rows=38535 width=17) (actual time=0.070..551.294 rows=859271 loops=1)`

```patch
+    AND payment.check_number IS NOT NULL
```
`->  Nested Loop  (cost=1.43..2577.12 rows=38535 width=17) (actual time=0.033..0.034 rows=0 loops=1)`

Reply:
Hi @fmdl 

any update here?